libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -771,6 +771,53 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	{
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		solAssert(!functionType->gasSet(), "Gas limit set for contract creation.");
+		solAssert(
+			functionType->returnParameterTypes().size() == 1,
+			"Constructor should return only one type"
+		);
+
+		TypePointers argumentTypes;
+		string constructorParams;
+		for (ASTPointer<Expression const> const& arg: arguments)
+		{
+			argumentTypes.push_back(arg->annotation().type);
+			constructorParams += ", " + IRVariable{*arg}.commaSeparatedList();
+		}
+
+		ContractDefinition const* contract =
+			&dynamic_cast<ContractType const&>(*functionType->returnParameterTypes().front()).contractDefinition();
+		m_context.objects().insert(contract);
+
+		Whiskers t(R"(
+			let <memPos> := <allocateUnbounded>(datasize("<object>"))
+			datacopy(<memPos>, dataoffset("<object>"), datasize("<object>"))
+			let	<memEnd> := <abiEncode>(add(<memPos>, datasize("<object>"))<constructorParams>)
+			<?saltSet>
+				let <retVars> := <?#salt>create2<!#salt>create</#salt>(<value>, <memPos>, sub(<memEnd>, <memPos>)<?#salt>, <salt></#salt>)
+			<!saltSet>
+				let <retVars> := create(<value>, <memPos>, sub(<memEnd>, <memPos>))
+			</saltSet>
+		)");
+		t("memPos", m_context.newYulVariable());
+		t("memEnd", m_context.newYulVariable());
+		t("allocateUnbounded", m_utils.allocationUnboundedFunction());
+		t("object", m_context.creationObjectName(*contract));
+		ABIFunctions abiFunctions(m_context.evmVersion(), m_context.revertStrings(), m_context.functionCollector());
```

Comment:
Maybe we could add a function to the context that returns the proper ABIFunctions object?

Reply:
Yes, and this is used on many places as well...